securedrop/journalist.py

```diff
@@ -333,6 +349,162 @@ def edit_account():
     return render_template('edit_account.html')
 
 
+def get_source_tags_in_use():
+    """Get only the source tags currently being used"""
+    source_tags = db_session.query(SourceTag, SourceLabelType) \
+                            .join(SourceLabelType) \
+                            .group_by(SourceLabelType.label_text).all()
+    return source_tags
+
+
+def get_all_defined_source_tags():
+    """Get all defined source tags"""
+    source_tags = db_session.query(SourceLabelType).all()
+    return source_tags
+
+
+def get_all_defined_submission_tags():
+    """Get all defined submission tags"""
+    submission_tags = db_session.query(SubmissionLabelType).all()
+    return submission_tags
+
+
+def get_source_tags(source_id):
+    source = Source.query.filter(Source.id == source_id).first()
+    source_tags = db_session.query(SourceTag, SourceLabelType) \
+                            .join(SourceLabelType) \
+                            .filter(SourceTag.source_id == source.id) \
+                            .group_by(SourceLabelType.label_text).all()
+    return source_tags
+
+
+def get_submission_tags(submission_id):
+    submission = Submission.query.filter(Submission.id == submission_id).first()
+    submission_tags = db_session.query(SubmissionTag, SubmissionLabelType) \
+                            .join(SubmissionLabelType) \
+                            .filter(SubmissionTag.submission_id == submission.id) \
+                            .group_by(SubmissionLabelType.label_text).all()
```

Comment:
Yep this was a bad query: removed the entire thing by making better use of SQLAlchemy's `relationship` directive in 476830e